### ISSUE TITLE
jQuery 4.0 release & integration tracking

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1896]
https://github.com/jquery/jquery/issues/5365#issuecomment-1819261993

Placeholder issue for jQuery 4.0.0 compatibility / issues.

**Comment by petecooper** [comment]
https://blog.jquery.com/2024/02/06/jquery-4-0-0-beta/

**Comment by bloatware** [comment]
Thanks @petecooper. It looks like we are hold by jQuery file uploader that uses few removed methods.

**Comment by bloatware** [comment]
My local txp runs jQuery 4 beta, so far so good. Should we try using it in dev, to enlarge the testers pool? We'd switch back to jQuery 3 of course if txp 4.9 is released before jQuery 4.

**Comment by bloatware** [comment]
Demo server is blocking something, not sure what.

**Comment by petecooper** [comment]
I can have a look - what can you see?

**Comment by petecooper** [comment]
>Refused to execute https://dev-demo.textpattern.co/dev/textpattern/index.php?_=1716559654637 as script because "X-Content-Type-Options: nosniff" was given and its Content-Type is not a script MIME type.

Might that be it?

**Comment by bloatware** [comment]
Yep, it looks like a csp conflict. Ironically, jQuery 4 [intends](https://github.com/jquery/jquery/issues/5476#issuecomment-2121312020) to solve them.

**Comment by petecooper** [comment]
I'll see how I can workaround it in the meantime, leave it with me for now.

**Comment by petecooper** [comment]
I'm a little perplexed - this appears to be solved, yet I haven't changed anything. Are you able to reproduce, @bloatware?

**Comment by bloatware** [comment]
@petecooper oops, sorry, I thought you have [seen](https://github.com/textpattern/textpattern/commit/bad4df504130f4218e41229464fec4a265884024). We'd rather do some tests locally before pushing it online.

**Comment by petecooper** [comment]
Now seen 👍

**Comment by phiw13** [comment]
I can of course reproduce the issue when the `X-Content-Type-Options "nosniff"` header is present. Once that is disabled (`header unset …` in `textpattern/.htaccess`), I still have saving issues on the Write tab (actually, no way to Save anything, old or new article) – both my old test site and a newish fairly default installation.

The Firefox console reports: 

As a warning
`The script from “http://txpdev.local/textpattern/index.php?_=1716781530549” was loaded even though its MIME type (“text/html”) is not a valid JavaScript MIME type.`

As an error:
`Uncaught SyntaxError: expected expression, got '<'` (index.php 1:1)

I suspect there is some attempt at setting a response header that fails

–^–

All other panels behave correctly as far as I can see, images/files are uploaded or replaced, categories created, preferences saved. The few admin-side plugins active do what they are supposed to… etc. This testing with a fairly strict set of CSP directives (in config.php)

**Comment by petecooper** [comment]
jQuery 4.0.0 beta 2 landed: https://blog.jquery.com/2024/07/17/second-beta-of-jquery-4-0-0/

**Comment by jools-r** [comment]
Just noting that with jQuery 4 beta 2, I'm seeing `m.isArray is not defined` errors in the console when trying to upload a file. @bloatware mentioned that probability above. The upload button then does nothing. Dropping back to the last version of jquery.js resolves it.

**Comment by bloatware** [comment]
I remember having patched the file uploader few months ago, but an update must have overridden it. Will probably repatch before the final 4.9 release, thanks for the report.

**Comment by petecooper** [comment]
4.0.0-rc.1 released:

https://github.com/jquery/jquery/releases/tag/4.0.0-rc.1

**Comment by petecooper** [comment]
https://blog.jquery.com/2026/01/17/jquery-4-0-0/

**Comment by petecooper** [comment]
I've updated `package.json` to retrieve jQuery 4.0.0 from npm – is there a way to force this 'get' action to test it out?

Tag @Bloke @jools-r @bloatware for any advice. Thank you.
